include/power_grid_model/math_solver/newton_raphson_pf_solver.hpp

```diff
@@ -213,117 +216,49 @@ constexpr block_entry_trait<PFJacBlock> jac_trait{};
-        DoubleComplex const u_ref = [&]() {
-            DoubleComplex sum_u_ref = 0.0;
-            for (Idx bus = 0; bus != n_bus_; ++bus) {
-                for (Idx source = source_bus_indptr[bus]; source != source_bus_indptr[bus + 1]; ++source) {
-                    sum_u_ref += input.source[source] * std::exp(1.0i * -phase_shift[bus]);  // offset phase shift
-                }
-            }
-            return sum_u_ref / (double)input.source.size();
-        }();
+    void initialize_unknown_polar(MathOutput<sym> output) {
         // assign u_ref as flat start
-        for (Idx i = 0; i != n_bus_; ++i) {
+        for (Idx i = 0; i != this->n_bus_; ++i) {
             // consider phase shift
-            output.u[i] = ComplexValue<sym>{u_ref * std::exp(1.0i * phase_shift[i])};
             x_[i].v = cabs(output.u[i]);
             x_[i].theta = arg(output.u[i]);
         }
-        sub_timer.stop();
-
-        // start calculation
-        // iteration
-        Idx num_iter = 0;
-        while (max_dev > err_tol) {
-            if (num_iter++ == max_iter) {
-                throw IterationDiverge{max_iter, max_dev, err_tol};
-            }
-            sub_timer = Timer(calculation_info, 2222, "Calculate jacobian and rhs");
-            calculate_jacobian_and_deviation(y_bus, input, output.u);
-            sub_timer = Timer(calculation_info, 2223, "Solve sparse linear equation");
-            bsr_solver_.solve(data_jac_.data(), del_pq_.data(), del_x_.data());
-            sub_timer = Timer(calculation_info, 2224, "Iterate unknown");
-            max_dev = iterate_unknown(output.u);
-            sub_timer.stop();
-        }
-
-        // calculate math result
-        sub_timer = Timer(calculation_info, 2225, "Calculate Math Result");
-        calculate_result(y_bus, input, output);
-
-        // Manually stop timers to avoid "Max number of iterations" to be included in the timing.
-        sub_timer.stop();
-        main_timer.stop();
-
-        const auto key = Timer::make_key(2226, "Max number of iterations");
-        calculation_info[key] = std::max(calculation_info[key], (double)num_iter);
-
-        return output;
     }
 
-   private:
-    Idx n_bus_;
-    // shared topo data
-    std::shared_ptr<DoubleVector const> phase_shift_;
-    std::shared_ptr<IdxVector const> load_gen_bus_indptr_;
-    std::shared_ptr<IdxVector const> source_bus_indptr_;
-    std::shared_ptr<std::vector<LoadGenType> const> load_gen_type_;
-    // data for jacobian
-    std::vector<PFJacBlock<sym>> data_jac_;
-    // calculation data
-    std::vector<PolarPhasor<sym>> x_;      // unknown
-    std::vector<PolarPhasor<sym>> del_x_;  // unknown iterative
-    // this stores in different steps
-    // 1. negative power injection: - p/q_calculated
-    // 2. power unbalance: p/q_specified - p/q_calculated
-    std::vector<ComplexPower<sym>> del_pq_;
-    BSRSolver<double> bsr_solver_;
+    void initialize_matrix(YBus<sym> const& y_bus) {
```

Comment:
you can just omit the name in the argument list (only the types)
then it will not issue unused variable warning.